book/tutorials/snowex-core/03_snowex_data_preview.ipynb

```diff
@@ -0,0 +1,822 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "36ef0083-ff1f-4e49-a978-920e6e55912e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# standard imports\n",
+    "import pandas as pd\n",
+    "import pyproj\n",
+    "\n",
+    "# plotting imports\n",
+    "import matplotlib.pyplot as plt\n",
+    "# from matplotlib import gridspec # do I use?\n",
+    "import matplotlib.dates as mdates\n",
+    "# import matplotlib.patches as patches # do I use?\n",
+    "import seaborn as sns\n",
+    "\n",
+    "# mapping imports\n",
+    "import folium\n",
+    "from pyproj import Transformer\n",
+    "import branca.colormap as cm\n",
+    "from folium import IFrame\n",
+    "from folium.plugins import MeasureControl\n",
+    "\n",
+    "# Earthaccess imports\n",
+    "import earthaccess"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb8ff033-1080-4798-bc96-de83b60da6bc",
+   "metadata": {},
+   "source": [
+    "Now we continue by accessing data with our EarthData accounts"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "824b209b-7f8a-4c43-9e6a-7d33d0766720",
+   "metadata": {},
+   "source": [
+    "### Download data from NASA's NSIDC DAAC  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dd5a71dd-80d8-42d7-93a9-f662aa1eff16",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# connect to Earthaccess\n",
+    "auth = earthaccess.login()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f4000c97-fc42-4171-8d95-5fb7bf94e4f0",
+   "metadata": {},
+   "source": [
+    "From [EarthData](https://search.earthdata.nasa.gov/search?fpj=SnowEx&fst0=cryosphere) we can glean information about the files we are interested in. For example, we learn that the SnowEx 2017 snow pit data are a multi-file granule, which will require one extra step to download. \n",
+    "![](./content/03_granules.png)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "21d6145a-aada-4104-a787-44c33c7350a8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pull SnowEx 2017 snow pit data - Summary_SWE file\n",
+    "results_gm17 =earthaccess.search_data(\n",
+    "    short_name='SNEX17_SNOWPITS',\n",
+    "    granule_name = 'SnowEx17_SnowPits_GM')\n",
+    "\n",
+    "# some extra code to filter a multi-file granule (2017 IOP granules: 1). SnowEx17_SnowPits_GM, 2). SnowEx17_SnowPits_SBB\n",
+    "filtered_urls = []\n",
+    "\n",
+    "for granule in results_gm17:\n",
+    "    links = granule.data_links()\n",
+    "    for link in links:\n",
+    "        if link == 'https://n5eil01u.ecs.nsidc.org/DP1/SNOWEX/SNEX17_SnowPits.001/2017.02.06/SnowEx17_SnowPits_GM_201702_v01.csv':\n",
+    "            filtered_urls.append(link) # Knowing the name of the file I wanted helped here\n",
+    "\n",
+    "earthaccess.download(filtered_urls, \n",
+    "                     local_path='./Downloads')"
```

Comment:
```suggestion
    "                     local_path='/tmp')"
```

This is erroring in CI, I suggest downloading things to `/tmp` instead of a local folder:

> ;31mFileNotFoundError[0m: [Errno 2] No such file or directory: './Downloads/SnowEx17_SnowPits_GM_201702_v01.csv'

Reply:
We also added this folder to the `.gitignore` and I am fine with leaving it there

Reply:
The actual error is the same as in the Data access tutorial here:
https://snowex-2024.hackweek.io/tutorials/Data_access/earthaccess_snowex.html#step-4-download-data-locally

You can see that in the [artifacts here](https://github.com/snowex-hackweek/website-2024/actions/runs/10426542661)

Reply:
hmm, yeah, i suspect you're right that the old creds aren't valid. I'll switch to ensure to using the new ones we know are working. Could add a step to actually check them 😅

```python
auth = earthaccess.login()
print(auth.authenticated)
```

Reply:
Great idea 👍 

What surprised me in the data access tutorial is that all the search call worked. It failed only upon downloading actually data. This makes me guess that the authentication failed

Reply:
yes - NASA search is anonymous, only download uses creds!

Reply:
OK, It's possible they were either outdated or I just pasted in the password incorrectly, but I just updated it and we're back in business